scout/adapter/mongo/clinvar.py

```diff
@@ -218,63 +218,55 @@ def sort_clinvar_case_data(self, variant_list, case_data_list):
 
         return sorted_case_data or case_data_list
 
-    def clinvar_submissions(self, institute_id):
-        """Collect all open and closed clinvar submissions for an institute
-
-        Args:
-            institute_id(str): an institute ID
+    def _basic_submission_info(self, result: dict) -> dict:
+        """Extracts the basic submission fields."""
+        user = self.user(user_id=result.get("created_by"))
+        return {
+            "_id": result.get("_id"),
+            "status": result.get("status"),
+            "institute_id": result.get("institute_id"),
+            "created_at": result.get("created_at"),
+            "created_by": user["name"] if user else None,
+            "updated_at": result.get("updated_at"),
+        }
 
-        Returns:
-            submissions(list): a list of clinvar submission objects
-        """
-        # get first all submission objects
+    def clinvar_submissions(self, institute_id: str) -> List[dict]:
+        """Collect all open and closed clinvar submissions for an institute"""
         query = dict(institute_id=institute_id)
         results = list(
             self.clinvar_submission_collection.find(query).sort("updated_at", pymongo.DESCENDING)
         )
 
         submissions = []
-        # Loop over all ClinVar submissions for an institute
         for result in results:
-            submission = {}
+            submission = self._basic_submission_info(result)
             cases = {}
-            user: dict = self.user(user_id=result.get("created_by"))
-            submission["_id"] = result.get("_id")
-            submission["status"] = result.get("status")
-            submission["institute_id"] = result.get("institute_id")
-            submission["created_at"] = result.get("created_at")
-            submission["created_by"] = user["name"] if user else None
-            submission["updated_at"] = result.get("updated_at")
-
-            if "clinvar_subm_id" in result:
+
+            if result.get("clinvar_subm_id"):
                 submission["clinvar_subm_id"] = result["clinvar_subm_id"]
 
-            # If submission has variants registered
```

Comment:
💯 
You could also make the docstring story slightly longer, stating the different possible outcomes, like eg the "case N/A" tag.

Reply:
I've added a specific comment to that line, this way it will be clearer I think

Reply:
All good, just a general principle for avoiding comments in the middle of code. 😸

Reply:
I know but on the line is easier to undertsand, and who reads the docstring anyway ;)